pandas/core/frame.py

```diff
@@ -3821,13 +3821,16 @@ def _apply_standard(self, func, axis, ignore_failures=False, reduce=True):
         # e.g. if we want to apply to a SparseFrame, then can't directly reduce
         if reduce:
 
-            try:
-
-                # the is the fast-path
-                values = self.values
-                dummy = Series(NA, index=self._get_axis(axis),
-                               dtype=values.dtype)
+            # the is the fast-path
+            values = self.values
+            # Create a dummy Series from an empty array
+            # Unlike filling with NA, this works for any dtype
+            index = self._get_axis(axis)
+            empty_arr = np.empty(len(index), dtype=values.dtype)
+            dummy = Series(empty_arr, index=self._get_axis(axis),
+                           dtype=values.dtype)
```

Comment:
Or I could fix `Series(index=..., dtype=int)` to return a series of 0's or something, but I would have to make sure there's a well-defined empty/zero value for any dtype.